platform/cosmos/api_test.go

```diff
@@ -232,41 +282,87 @@ var transferDst = blockatlas.Tx{
 	Date:   1556992677,
 	Block:  151980,
 	Status: blockatlas.StatusCompleted,
+	Type:   blockatlas.TxTransfer,
 	Meta: blockatlas.Transfer{
 		Value:    "2271999999",
 		Symbol:   "ATOM",
 		Decimals: 6,
 	},
 }
 
-func TestNormalize(t *testing.T) {
-	testNormalize(t, transferSrc, &transferDst)
-	testNormalize(t, delegateSrc, &delegateDst)
-	testNormalize(t, unDelegateSrc, &unDelegateDst)
+var delegateDst = blockatlas.Tx{
+	ID:     "11078091D1D5BD84F4275B6CEE02170428944DB0E8EEC37E980551435F6D04C7",
+	Coin:   coin.ATOM,
+	From:   "cosmos1237l0vauhw78qtwq045jd24ay4urpec6r3xfn3",
+	To:     "cosmosvaloper12w6tynmjzq4l8zdla3v4x0jt8lt4rcz5gk7zg2",
+	Fee:    "5000",
+	Date:   1564632616,
+	Block:  1258202,
+	Status: blockatlas.StatusCompleted,
+	Type:   blockatlas.TxAnyAction,
+	Meta: blockatlas.AnyAction{
+		Coin:     coin.ATOM,
+		Title:    blockatlas.AnyActionDelegation,
+		Key:      blockatlas.KeyStakeDelegate,
+		Name:     "ATOM",
+		Symbol:   coin.Coins[coin.ATOM].Symbol,
+		Decimals: coin.Coins[coin.ATOM].Decimals,
+		Value:    "49920",
+	},
 }
 
-func testNormalize(t *testing.T, src string, dst *blockatlas.Tx) {
-	var srcTx Tx
-	err := json.Unmarshal([]byte(src), &srcTx)
-	if err != nil {
-		t.Error(err)
-		return
-	}
-
-	tx, _ := Normalize(&srcTx)
-	resJSON, err := json.Marshal(&tx)
-	if err != nil {
-		t.Fatal(err)
-	}
+var unDelegateDst = blockatlas.Tx{
+	ID:     "A1EC36741FEF681F4A77B8F6032AD081100EE5ECB4CC76AEAC2174BC6B871CFE",
+	Coin:   coin.ATOM,
+	From:   "cosmos137rrp4p8n0nqcft0mwc62tdnyhhzf80knv5t94",
+	To:     "cosmosvaloper1te8nxpc2myjfrhaty0dnzdhs5ahdh5agzuym9v",
+	Fee:    "5000",
+	Date:   1564624521,
+	Block:  1257037,
+	Status: blockatlas.StatusCompleted,
+	Type:   blockatlas.TxAnyAction,
+	Meta: blockatlas.AnyAction{
+		Coin:     coin.ATOM,
+		Title:    blockatlas.AnyActionUndelegation,
+		Key:      blockatlas.KeyStakeDelegate,
+		Name:     "ATOM",
+		Symbol:   coin.Coins[coin.ATOM].Symbol,
+		Decimals: coin.Coins[coin.ATOM].Decimals,
+		Value:    "5100000000",
+	},
+}
+var failedTransferDst = blockatlas.Tx{
+	ID:     "5E78C65A8C1A6C8239EBBBBF2E42020E6ADBA8037EDEA83BF88E1A9159CF13B8",
+	Coin:   coin.ATOM,
+	From:   "cosmos1shpfyt7psrff2ux7nznxvj6f7gq59fcqng5mku",
+	To:     "cosmos1za4pu5gxm80fg6sx0956f88l2sx7jfg2vf7nlc",
+	Fee:    "2000",
+	Date:   1576120902,
+	Block:  5552,
+	Status: blockatlas.StatusFailed,
+	Type:   blockatlas.TxTransfer,
+	Memo:   "UniCoins registration rewards",
+	Meta: blockatlas.Transfer{
+		Value:    "100000",
+		Symbol:   "ATOM",
+		Decimals: 6,
+	},
+}
 
-	dstJSON, err := json.Marshal(&dst)
-	if err != nil {
-		t.Fatal(err)
-	}
+func TestNormalize(t *testing.T) {
+	testNormalize(t, "test transfer tx", transferSrc, transferDst)
```

Comment:
lets use table tests

Reply:
In this case don't make a lot of sense, because the test is already dynamic. But I did the changes.

Reply:
one good example of table tests changes

https://github.com/trustwallet/backend/pull/271/files#diff-aecf3ca183be94d51ad0c89054705d7d